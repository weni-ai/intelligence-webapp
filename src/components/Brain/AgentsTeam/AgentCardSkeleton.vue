<template>
  <section class="agent-card-skeleton">
    <UnnnicSkeletonLoading
      tag="div"
      width="30%"
      height="22px"
    />

    <UnnnicSkeletonLoading
      tag="div"
      width="100%"
      height="44px"
    />

    <section class="agent-card-skeleton__skills">
      <UnnnicSkeletonLoading
        tag="div"
        width="100px"
        height="30px"
      />
      <UnnnicSkeletonLoading
        tag="div"
        width="150px"
        height="30px"
      />
    </section>

    <UnnnicSkeletonLoading
      tag="div"
      width="100%"
      height="38px"
    />
  </section>
</template>

<style lang="scss" scoped>
.agent-card-skeleton {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-xs;

  &__skills {
    width: 100%;
    display: flex;
    gap: $unnnic-spacing-nano;
    flex-wrap: wrap;
  }
}
</style>
