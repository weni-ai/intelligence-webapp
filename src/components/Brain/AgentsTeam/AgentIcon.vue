<template>
  <img
    v-if="icon && agentIcon"
    :src="agentIcon"
    class="content__icon"
    data-testid="agent-icon"
  />
</template>

<script setup>
import { computed } from 'vue';
import { agentIcons } from '@/utils/agentIconService';

const props = defineProps({
  icon: {
    type: String,
    default: '',
  },
});

const agentIcon = computed(() => {
  if (!props.icon) return;

  return agentIcons[props.icon] || null;
});
</script>
