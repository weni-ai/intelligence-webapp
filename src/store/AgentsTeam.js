import { defineStore } from 'pinia';
import { computed, reactive, ref } from 'vue';

import nexusaiAPI from '@/api/nexusaiAPI.js';
import { useAlertStore } from './Alert';
import agentIconService from '@/utils/agentIconService';

import i18n from '@/utils/plugins/i18n';

export const useAgentsTeamStore = defineStore('AgentsTeam', () => {
  const linkToCreateAgent = 'https://github.com/weni-ai/weni-cli';

  const alertStore = useAlertStore();

  const activeTeam = reactive({
    status: null,
    data: {
      manager: null,
      agents: [],
    },
  });

  const officialAgents = reactive({
    status: null,
    data: [],
  });

  const myAgents = reactive({
    status: null,
    data: [],
  });

  const allAgents = computed(() => {
    return {
      manager: activeTeam.data.manager,
      agents: [...officialAgents.data, ...myAgents.data],
    };
  });

  const isAgentsGalleryOpen = ref(false);

  async function loadActiveTeam() {
    try {
      activeTeam.status = 'loading';

      const { data } = await nexusaiAPI.router.agents_team.listActiveTeam();

      activeTeam.data = {
        manager: data.manager,
        agents: agentIconService.applyIconsToAgents(data.agents),
      };
      activeTeam.status = 'complete';
    } catch (error) {
      console.error('error', error);

      activeTeam.status = 'error';
    }
  }

  async function loadOfficialAgents({ search = '' } = {}) {
    try {
      officialAgents.status = 'loading';

      const { data } = await nexusaiAPI.router.agents_team.listOfficialAgents({
        search,
      });

      officialAgents.data = agentIconService.applyIconsToAgents(data);
      officialAgents.status = 'complete';
    } catch (error) {
      console.error('error', error);

      officialAgents.status = 'error';
    }
  }

  async function loadMyAgents({ search = '' } = {}) {
    try {
      myAgents.status = 'loading';

      const { data } = await nexusaiAPI.router.agents_team.listMyAgents({
        search,
      });

      myAgents.data = agentIconService.applyIconsToAgents(data);
      myAgents.status = 'complete';
    } catch (error) {
      console.error('error', error);

      myAgents.status = 'error';
    }
  }

  async function toggleAgentAssignment({ uuid, is_assigned }) {
    if (!uuid || typeof is_assigned !== 'boolean') {
      throw new Error('uuid and is_assigned are required');
    }

    try {
      const agent =
        officialAgents.data.find((agent) => agent.uuid === uuid) ||
        myAgents.data.find((agent) => agent.uuid === uuid);

      if (!agent) {
        throw new Error('Agent not found');
      }

      const { data } =
        await nexusaiAPI.router.agents_team.toggleAgentAssignment({
          agentUuid: agent?.uuid,
          is_assigned,
        });

      agent.assigned = data.assigned;

      if (is_assigned) {
        activeTeam.data.agents.push(agent);
      } else {
        activeTeam.data.agents = activeTeam.data.agents.filter(
          (agent) => agent.uuid !== uuid,
        );
      }

      alertStore.add({
        text: i18n.global.t(
          is_assigned
            ? 'router.agents_team.card.success_assign_alert'
            : 'router.agents_team.card.success_unassign_alert',
          {
            agent: agent.name,
          },
        ),
        type: 'success',
      });

      return {
        status: 'success',
      };
    } catch (error) {
      console.error('error', error);

      alertStore.add({
        text: i18n.global.t('router.agents_team.card.error_alert'),
        type: 'error',
      });

      return {
        status: 'error',
      };
    }
  }

  return {
    linkToCreateAgent,
    activeTeam,
    officialAgents,
    myAgents,
    allAgents,
    isAgentsGalleryOpen,
    loadActiveTeam,
    loadOfficialAgents,
    loadMyAgents,
    toggleAgentAssignment,
  };
});
