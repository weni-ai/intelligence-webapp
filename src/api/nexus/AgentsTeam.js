import { computed } from 'vue';
import request from '@/api/nexusaiRequest';
import { useProjectStore } from '@/store/Project';
import { cleanParams } from '@/utils/http';

const projectUuid = computed(() => useProjectStore().uuid);

export const AgentsTeam = {
  async listOfficialAgents({ search }) {
    const params = cleanParams({
      search,
    });
    const { data } = await request.$http.get(
      `api/agents/official/${projectUuid.value}`,
      {
        params,
      },
    );

    return {
      data: data.map(
        ({
          uuid,
          name,
          description,
          skills,
          assigned,
          credentials,
          is_official,
          slug,
        }) => ({
          uuid,
          name,
          description,
          skills,
          assigned,
          credentials,
          is_official,
          id: slug,
        }),
      ),
    };
  },

  async listMyAgents({ search }) {
    const params = cleanParams({
      search,
    });
    const { data } = await request.$http.get(
      `api/agents/my-agents/${projectUuid.value}`,
      {
        params,
      },
    );

    return {
      data: data.map(
        ({
          uuid,
          name,
          description,
          skills,
          assigned,
          credentials,
          is_official,
          slug,
        }) => ({
          uuid,
          name,
          description,
          skills,
          assigned,
          credentials,
          is_official,
          id: slug,
        }),
      ),
    };
  },

  async listActiveTeam() {
    const { data } = await request.$http.get(
      `api/agents/teams/${projectUuid.value}`,
    );

    const { manager, agents } = data;

    return {
      data: {
        manager: {
          id: manager.id || 'manager',
        },
        agents: agents.map(
          ({
            uuid,
            name,
            skills,
            id,
            description,
            credentials,
            is_official,
          }) => ({
            uuid,
            name,
            skills,
            id,
            description,
            credentials,
            is_official,
          }),
        ),
      },
    };
  },

  async toggleAgentAssignment({ agentUuid, is_assigned }) {
    const { data } = await request.$http.patch(
      `api/project/${projectUuid.value}/assign/${agentUuid}`,
      {
        assigned: is_assigned,
      },
      {
        hideGenericErrorAlert: true,
      },
    );

    return {
      data,
    };
  },
};
