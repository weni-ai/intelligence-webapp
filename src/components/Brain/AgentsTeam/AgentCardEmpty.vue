<template>
  <button
    class="agent-card-empty"
    data-testid="agent-card-empty"
    @click="createNewAgent"
  >
    <UnnnicIcon
      icon="add"
      size="md"
      scheme="neutral-clean"
      data-testid="agent-card-empty-icon"
    />

    <UnnnicIntelligenceText
      tag="p"
      family="secondary"
      size="body-gt"
      color="neutral-dark"
      data-testid="agent-card-empty-title"
    >
      {{ $t('router.agents_team.create_agent') }}
    </UnnnicIntelligenceText>
  </button>
</template>

<script setup>
import { useAgentsTeamStore } from '@/store/AgentsTeam';

const agentsTeamStore = useAgentsTeamStore();
function createNewAgent() {
  window.open(agentsTeamStore.linkToCreateAgent, '_blank');
}
</script>

<style lang="scss" scoped>
.agent-card-empty {
  border: none;
  background: none;

  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;

  height: 100%;
  width: 100%;

  cursor: pointer;
}
</style>
