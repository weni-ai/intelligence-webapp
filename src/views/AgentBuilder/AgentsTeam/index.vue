<template>
  <section class="agents-team">
    <AgentBuilderHeader
      title="Agents Team"
      subtitle="Assign agents to your team"
      withDivider
      actionsSize="lg"
    >
      <template #actions>
        <section class="agents-team__actions">
          <UnnnicButton
            type="secondary"
            iconLeft="add"
            @click="handleAgentsGallery"
          >
            {{ $t('router.agents_team.assign_agents') }}
          </UnnnicButton>

          <UnnnicButton
            type="primary"
            iconLeft="play_arrow"
            iconsFilled
            @click="handlePreview"
          >
            {{ $t('router.agents_team.preview') }}
          </UnnnicButton>
        </section>
      </template>
    </AgentBuilderHeader>

    <ActiveTeam />

    <AgentsGalleryModal />

    <PreviewDrawer v-model="isPreviewOpen" />
  </section>
</template>

<script setup>
import { ref } from 'vue';

import AgentBuilderHeader from '@/components/AgentBuilder/Header.vue';
import PreviewDrawer from '@/components/Brain/Preview/PreviewDrawer.vue';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

import ActiveTeam from './ActiveTeam.vue';
import AgentsGalleryModal from './AgentsGalleryModal.vue';

const isPreviewOpen = ref(false);

const handleAgentsGallery = () => {
  useAgentsTeamStore().isAgentsGalleryOpen = true;
};

const handlePreview = () => {
  isPreviewOpen.value = true;
};
</script>

<style lang="scss" scoped>
section.agents-team {
  display: grid;
  grid-template-rows: auto 1fr;

  .agents-team__actions {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: $unnnic-spacing-sm;
  }
}
</style>
