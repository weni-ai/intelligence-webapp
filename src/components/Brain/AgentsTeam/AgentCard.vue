<template>
  <section
    data-testid="agent-card"
    :class="['agent-card', { 'agent-card--empty': empty }]"
  >
    <AgentCardSkeleton
      v-if="loading"
      data-testid="agent-card-skeleton"
    />

    <AgentCardEmpty
      v-else-if="empty"
      data-testid="agent-card-empty"
    />

    <section
      v-else
      class="agent-card__content"
    >
      <UnnnicIntelligenceText
        tag="p"
        family="secondary"
        size="body-gt"
        color="neutral-darkest"
        weight="bold"
        data-testid="title"
      >
        {{ title }}
      </UnnnicIntelligenceText>

      <UnnnicIntelligenceText
        v-if="description"
        tag="p"
        family="secondary"
        size="body-gt"
        color="neutral-cloudy"
        data-testid="description"
      >
        {{ description }}
      </UnnnicIntelligenceText>

      <section class="content__skills">
        <Skill
          v-for="skill in skills"
          :key="skill.name"
          :title="skill.name"
          :icon="skill.icon"
          data-testid="skill"
        />
      </section>
    </section>

    <UnnnicButton
      v-if="!loading && !empty && assignment"
      :class="[
        'agent-card__button',
        { 'agent-card__button--assigned': assigned },
      ]"
      :text="
        assigned
          ? $t('router.agents_team.card.assigned')
          : $t('router.agents_team.card.assign')
      "
      :type="assigned ? 'tertiary' : 'primary'"
      :iconLeft="assigned ? 'check' : ''"
      size="small"
      :loading="isAssigning"
      data-testid="assign-button"
      @click="toggleAgentAssignment"
    />
  </section>
</template>

<script setup>
import { ref } from 'vue';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

import AgentCardSkeleton from './AgentCardSkeleton.vue';
import AgentCardEmpty from './AgentCardEmpty.vue';
import Skill from './Skill.vue';

defineEmits(['assign']);

const props = defineProps({
  loading: {
    type: Boolean,
    default: false,
  },
  title: {
    type: String,
    default: '',
  },
  description: {
    type: String,
    default: '',
  },
  /**
   * An array of skills where each skill is an object with the following properties:
   * @property {string} name - The name of the skill
   * @property {string} icon - The icon of the skill
   */
  skills: {
    type: Array,
    default: () => [],
    validator(skills) {
      return skills.every((skill) => 'name' in skill && 'icon' in skill);
    },
  },
  assignment: {
    type: Boolean,
    default: true,
  },
  assigned: {
    type: Boolean,
    default: false,
  },
  uuid: {
    type: String,
    default: '',
  },
  empty: {
    type: Boolean,
    default: false,
  },
});

const isAssigning = ref(false);
const agentsTeamStore = useAgentsTeamStore();

async function toggleAgentAssignment() {
  isAssigning.value = true;

  try {
    await agentsTeamStore.toggleAgentAssignment({
      uuid: props.uuid,
      is_assigned: !props.assigned,
    });
  } catch (error) {
    console.error(error);
  } finally {
    isAssigning.value = false;
  }
}
</script>

<style lang="scss" scoped>
.agent-card {
  border-radius: $unnnic-border-radius-md;
  border: $unnnic-border-width-thinner solid $unnnic-color-neutral-cleanest;

  padding: $unnnic-spacing-sm;

  display: grid;
  gap: $unnnic-spacing-sm;
  align-content: space-between;

  &--empty {
    display: flex;

    cursor: pointer;
  }

  &__content {
    display: flex;
    flex-direction: column;
    gap: $unnnic-spacing-xs;

    .content__skills {
      display: flex;
      gap: $unnnic-spacing-nano;
      flex-wrap: wrap;
    }
  }

  :deep(.unnnic-button).agent-card__button {
    &--assigned {
      color: $unnnic-color-weni-600;
    }

    [class*='icon'] {
      color: $unnnic-color-weni-600;
    }
  }
}
</style>
