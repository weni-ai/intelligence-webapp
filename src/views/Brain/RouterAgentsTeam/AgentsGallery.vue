<template>
  <section
    class="agents-gallery"
    data-testid="agents-gallery"
  >
    <UnnnicIntelligenceText
      tag="h2"
      family="secondary"
      size="body-lg"
      color="neutral-darkest"
      weight="bold"
    >
      {{ $t('router.agents_team.gallery.title') }}
    </UnnnicIntelligenceText>

    <UnnnicTab
      class="agents-gallery__tabs"
      :tabs="contentTabs.map((tab) => tab.page)"
      :activeTab="activeTab"
      data-testid="agents-gallery-tabs"
      @change="onTabChange"
    >
      <template
        v-for="tab in contentTabs"
        :key="tab.page"
        #[`tab-head-${tab.page}`]
      >
        {{ $t(`router.agents_team.gallery.${tab.title}`) }}
      </template>
    </UnnnicTab>

    <UnnnicInput
      v-model="search[activeTab]"
      iconLeft="search"
      :placeholder="$t('router.agents_team.gallery.search_placeholder')"
      data-testid="search-input"
    />

    <section
      :class="[
        'agents-gallery__cards',
        {
          'agents-gallery__cards--empty': isMyAgentsEmpty,
        },
      ]"
      data-testid="agent-cards"
    >
      <template v-if="isLoadingAgents">
        <AgentCard
          v-for="(_, index) in Array(3)"
          :key="index"
          loading
          data-testid="agent-card-loading"
        />
      </template>

      <template v-else>
        <UnnnicIntelligenceText
          v-if="isSearchEmpty"
          color="neutral-clean"
          family="secondary"
          size="body-gt"
          tag="p"
        >
          {{ $t('router.agents_team.gallery.no_agent_found') }}
        </UnnnicIntelligenceText>

        <template v-else>
          <AgentCard
            v-if="activeTab === 'my-agents'"
            data-testid="agent-card-empty"
            empty
          />

          <AgentCard
            v-for="agent in agentsData"
            :key="agent.uuid"
            :title="agent.name"
            :description="agent.description"
            :skills="agent.skills"
            :uuid="agent.uuid"
            :assigned="agent.assigned"
            data-testid="agent-card"
          />
        </template>
      </template>
    </section>
  </section>
</template>

<script setup>
import { computed, onMounted, ref, watch } from 'vue';
import { debounce } from 'lodash';

import AgentCard from '@/components/Brain/AgentsTeam/AgentCard.vue';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

const agentsTeamStore = useAgentsTeamStore();
const officialAgents = agentsTeamStore.officialAgents;
const myAgents = agentsTeamStore.myAgents;

const contentTabs = ref([
  { title: 'official', page: 'official' },
  { title: 'my-agents', page: 'my-agents' },
]);
const activeTab = ref(contentTabs.value[0].page);
const search = ref({
  official: '',
  'my-agents': '',
});
const isSearchEmpty = ref(false);

const agentsData = computed(() =>
  activeTab.value === 'official' ? officialAgents.data : myAgents.data,
);
const isLoadingAgents = computed(
  () => officialAgents.status === 'loading' || myAgents.status === 'loading',
);
const isMyAgentsEmpty = computed(
  () =>
    activeTab.value === 'my-agents' &&
    myAgents.data.length === 0 &&
    !isLoadingAgents.value,
);

function updateSearchEmpty() {
  const currentAgents = agentsData.value;
  const currentSearch = search.value[activeTab.value];

  isSearchEmpty.value =
    currentSearch.trim() !== '' && currentAgents.length === 0;
}

function onTabChange(newTab) {
  activeTab.value = newTab;

  if (agentsTeamStore.myAgents.status === null && newTab === 'my-agents') {
    agentsTeamStore.loadMyAgents();
  }

  updateSearchEmpty();
}

onMounted(() => {
  agentsTeamStore.loadOfficialAgents();
});

const debouncedSearch = (callback) => debounce(callback, 300);
watch(
  () => search.value['my-agents'],
  debouncedSearch(async (search) => {
    await agentsTeamStore.loadMyAgents({ search });
    updateSearchEmpty();
  }),
);
watch(
  () => search.value['official'],
  debouncedSearch(async (search) => {
    await agentsTeamStore.loadOfficialAgents({ search });
    updateSearchEmpty();
  }),
);
</script>

<style lang="scss" scoped>
.agents-gallery {
  display: flex;
  flex-direction: column;
  gap: $unnnic-spacing-sm;

  &__tabs {
    margin-bottom: -$unnnic-spacing-sm;
  }

  &__cards {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    gap: $unnnic-spacing-sm;

    &--empty {
      grid-template-columns: repeat(1, 1fr);
    }
  }
}
</style>
