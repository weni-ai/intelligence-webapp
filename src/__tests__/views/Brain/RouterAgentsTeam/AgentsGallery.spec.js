import { afterEach, beforeEach, expect } from 'vitest';
import { nextTick } from 'vue';
import { shallowMount } from '@vue/test-utils';
import { createTestingPinia } from '@pinia/testing';

import { useAgentsTeamStore } from '@/store/AgentsTeam';

import AgentsGallery from '@/views/Brain/RouterAgentsTeam/AgentsGallery.vue';

const pinia = createTestingPinia({
  initialState: {
    AgentsTeam: {
      officialAgents: {
        status: null,
        data: [
          {
            uuid: 'uuid-1',
            name: 'Agent 1',
            description: 'Description 1',
            skills: [{ name: 'Skill 1', icon: 'ðŸ›’' }],
          },
          {
            uuid: 'uuid-2',
            name: 'Agent 2',
            description: 'Description 2',
            skills: [{ name: 'Skill 2', icon: 'ðŸ›’' }],
          },
        ],
      },
      myAgents: {
        status: null,
        data: [
          {
            uuid: 'uuid-3',
            name: 'My Agent 1',
            description: 'My Description 1',
            skills: [{ name: 'Skill 3', icon: 'ðŸ›’' }],
          },
          {
            uuid: 'uuid-4',
            name: 'My Agent 2',
            description: 'My Description 2',
            skills: [{ name: 'Skill 4', icon: 'ðŸ›’' }],
          },
        ],
      },
    },
  },
});

describe('AgentsGallery.vue', () => {
  let wrapper;
  const agentsTeamStore = useAgentsTeamStore();

  beforeEach(() => {
    wrapper = shallowMount(AgentsGallery, {
      global: {
        plugins: [pinia],
      },
    });
  });

  afterEach(() => {
    vi.clearAllMocks();
  });

  it('should match snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should renders the component', () => {
    expect(wrapper.find('[data-testid="agents-gallery"]').exists()).toBe(true);
  });

  it('should renders the tabs', () => {
    const tabs = wrapper.findComponent('[data-testid="agents-gallery-tabs"]');

    expect(tabs.props('tabs').length).toBe(2);
    expect(tabs.props('activeTab')).toBe('official');
  });

  it('should renders the search input', () => {
    const searchInput = wrapper.find('[data-testid="search-input"]');
    expect(searchInput.exists()).toBe(true);
  });

  it('should renders the agent cards', () => {
    const agentCards = wrapper.findAllComponents('[data-testid="agent-card"]');

    expect(agentCards.length).toBe(2);
  });

  it('should renders the loading state', async () => {
    agentsTeamStore.officialAgents.status = 'loading';
    await nextTick();

    const loadingCards = wrapper.findAll('[data-testid="agent-card-loading"]');
    expect(loadingCards.length).toBe(3);
  });

  it('should change the data when tab changes', async () => {
    expect(wrapper.vm.agentsData).toBe(agentsTeamStore.officialAgents.data);

    wrapper.vm.activeTab = 'my-agents';
    await nextTick();

    expect(wrapper.vm.agentsData).toBe(agentsTeamStore.myAgents.data);
  });

  it('should load my agents when tab changes if not loaded', async () => {
    agentsTeamStore.myAgents.status = null;
    wrapper.vm.onTabChange('my-agents');
    await nextTick();

    expect(agentsTeamStore.loadMyAgents).toHaveBeenCalledTimes(1);
    agentsTeamStore.myAgents.status = 'complete';

    wrapper.vm.onTabChange('official');
    wrapper.vm.onTabChange('my-agents');
    await nextTick();

    expect(agentsTeamStore.loadMyAgents).toHaveBeenCalledTimes(1);
  });

  it('should search agents when search input changes', async () => {
    vi.useFakeTimers();

    vi.clearAllMocks();
    wrapper.vm.search['official'] = 'Test Search';
    await vi.runAllTimers();

    expect(agentsTeamStore.loadOfficialAgents).toHaveBeenCalledTimes(1);

    wrapper.vm.search['my-agents'] = 'Test Search';
    await vi.runAllTimers();
    expect(agentsTeamStore.loadMyAgents).toHaveBeenCalledTimes(1);

    vi.useRealTimers();
  });
});
