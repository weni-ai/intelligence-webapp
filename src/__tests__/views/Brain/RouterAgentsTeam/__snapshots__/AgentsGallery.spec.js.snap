// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AgentsGallery.vue > should match snapshot 1`] = `
<section
  class="unnnic-modal-dialog agents-gallery-modal"
  data-testid="modal-dialog"
  data-v-4005123d=""
  data-v-ab1febba=""
>
  <section
    class="unnnic-modal-dialog__overlay"
    data-testid="modal-overlay"
    data-v-ab1febba=""
  />
  <section
    class="unnnic-modal-dialog__container unnnic-modal-dialog__container--lg"
    data-testid="modal-container"
    data-v-ab1febba=""
  >
    <!---->
    <section
      class="unnnic-modal-dialog__container__body"
      data-v-ab1febba=""
    >
      <header
        class="unnnic-modal-dialog__container__header"
        data-v-ab1febba=""
      >
        <section
          class="unnnic-modal-dialog__container__title-container"
          data-v-ab1febba=""
        >
          <!---->
          <h1
            class="unnnic-modal-dialog__container__title-text"
            data-testid="title-text"
            data-v-ab1febba=""
          >
            Agents gallery
          </h1>
        </section>
        <span
          class="material-symbols-rounded unnnic-icon-scheme--neutral-darkest unnnic-icon-size--md unnnic--clickable"
          data-testid="close-icon"
          data-v-9846e6db=""
          data-v-ab1febba=""
          translate="no"
        >
          close
        </span>
      </header>
      <section
        class="unnnic-modal-dialog__container__content"
        data-v-ab1febba=""
      >
        
        <section
          class="agents-gallery"
          data-testid="agents-gallery"
          data-v-4005123d=""
        >
          <div
            class="unnnic-side-bar-menu agents-gallery__sidebar"
            data-testid="sidebar-menu"
            data-v-4005123d=""
          >
            <p
              class="unnnic-side-bar-menu__title"
            />
            
            
            <section
              class="side-bar-item side-bar-item--active unnnic--clickable sidebar__item"
              data-testid="sidebar-item"
              data-v-4005123d=""
            >
              <!---->
              <p
                class="side-bar-item__label"
              >
                Official agents
              </p>
              <!--v-if-->
            </section>
            <section
              class="side-bar-item unnnic--clickable sidebar__item"
              data-testid="sidebar-item"
              data-v-4005123d=""
            >
              <!---->
              <p
                class="side-bar-item__label"
              >
                Custom agents
              </p>
              <!--v-if-->
            </section>
            
            
          </div>
          <section
            class="agents-gallery__content"
            data-v-4005123d=""
          >
            <div
              class="unnnic-form md"
              data-testid="search-input"
              data-v-29b8feb1=""
              data-v-4005123d=""
            >
              <!---->
              <div
                class="text-input size--md text-input--icon-right-size-sm unnnic-form-input"
                data-testid="search-input"
                data-v-29b8feb1=""
                data-v-c1401a33=""
                mask=""
              >
                <input
                  class="unnnic-form-input input-itself input size-md normal input--has-icon-left unnnic-form-input input-itself"
                  data-testid="search-input"
                  data-v-244e8315=""
                  data-v-c1401a33=""
                  hascloudycolor="false"
                  iconleft="search"
                  iconrightsize="sm"
                  placeholder="Search by agent or skill"
                  type="text"
                  value=""
                />
                <span
                  class="material-symbols-rounded unnnic-icon-scheme--neutral-cloudy unnnic-icon-size--sm icon-left"
                  data-testid="material-icon"
                  data-v-9846e6db=""
                  data-v-c1401a33=""
                  translate="no"
                >
                  search
                </span>
                <!---->
              </div>
              <!---->
            </div>
            <!--v-if-->
            <section
              class="content__cards"
              data-testid="agent-cards"
              data-v-4005123d=""
            >
              
              
              
              <section
                class="assign-agent-card"
                data-testid="assign-agent-card"
                data-v-73104bd4=""
              >
                <section
                  class="assign-agent-card__content"
                  data-v-73104bd4=""
                >
                  <header
                    class="content__header"
                    data-v-73104bd4=""
                  >
                    <p
                      class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-bold unnnic-text--color-neutral-darkest"
                      data-testid="title"
                      data-v-73104bd4=""
                      data-v-ec8da4ad=""
                    >
                      
                      Agent 1
                      
                    </p>
                    <!--v-if-->
                  </header>
                  <p
                    class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy content__description content__description"
                    data-testid="description"
                    data-v-73104bd4=""
                    data-v-ec8da4ad=""
                  >
                    
                    Description 1
                    
                  </p>
                  <section
                    class="content__skills"
                    data-v-73104bd4=""
                  >
                    
                    <section
                      class="skill"
                      data-testid="skill"
                      data-v-29c532b5=""
                      data-v-73104bd4=""
                    >
                      <p
                        class="skill__icon"
                        data-testid="skill-icon"
                        data-v-29c532b5=""
                      >
                        ðŸ›’
                      </p>
                      <p
                        class="unnnic-text unnnic-text--size-body-md unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy skill__title skill__title"
                        data-testid="skill-name"
                        data-v-29c532b5=""
                        data-v-ec8da4ad=""
                      >
                        
                        Skill 1
                        
                      </p>
                    </section>
                    
                  </section>
                </section>
                <button
                  class="assign-agent-card__button unnnic-button unnnic-button--size-small unnnic-button--primary assign-agent-card__button"
                  data-testid="assign-button"
                  data-v-73104bd4=""
                  data-v-d127031e=""
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button__label"
                    data-testid="button-label"
                    data-v-d127031e=""
                  >
                    
                    
                     Assign to team
                  </span>
                  <!---->
                </button>
              </section>
              <!--v-if-->
              
              
              <section
                class="assign-agent-card"
                data-testid="assign-agent-card"
                data-v-73104bd4=""
              >
                <section
                  class="assign-agent-card__content"
                  data-v-73104bd4=""
                >
                  <header
                    class="content__header"
                    data-v-73104bd4=""
                  >
                    <p
                      class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-bold unnnic-text--color-neutral-darkest"
                      data-testid="title"
                      data-v-73104bd4=""
                      data-v-ec8da4ad=""
                    >
                      
                      Agent 2
                      
                    </p>
                    <!--v-if-->
                  </header>
                  <p
                    class="unnnic-text unnnic-text--size-body-gt unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy content__description content__description"
                    data-testid="description"
                    data-v-73104bd4=""
                    data-v-ec8da4ad=""
                  >
                    
                    Description 2
                    
                  </p>
                  <section
                    class="content__skills"
                    data-v-73104bd4=""
                  >
                    
                    <section
                      class="skill"
                      data-testid="skill"
                      data-v-29c532b5=""
                      data-v-73104bd4=""
                    >
                      <p
                        class="skill__icon"
                        data-testid="skill-icon"
                        data-v-29c532b5=""
                      >
                        ðŸ›’
                      </p>
                      <p
                        class="unnnic-text unnnic-text--size-body-md unnnic-text--line-height-md unnnic-text--family-secondary unnnic-text--weight-regular unnnic-text--color-neutral-cloudy skill__title skill__title"
                        data-testid="skill-name"
                        data-v-29c532b5=""
                        data-v-ec8da4ad=""
                      >
                        
                        Skill 2
                        
                      </p>
                    </section>
                    
                  </section>
                </section>
                <button
                  class="assign-agent-card__button unnnic-button unnnic-button--size-small unnnic-button--primary assign-agent-card__button"
                  data-testid="assign-button"
                  data-v-73104bd4=""
                  data-v-d127031e=""
                >
                  <!---->
                  <!---->
                  <!---->
                  <span
                    class="unnnic-button__label"
                    data-testid="button-label"
                    data-v-d127031e=""
                  >
                    
                    
                     Assign to team
                  </span>
                  <!---->
                </button>
              </section>
              <!--v-if-->
              
              
              
            </section>
          </section>
        </section>
        
      </section>
      <!---->
    </section>
  </section>
</section>
`;
